__tests__/example.test.res.type.ts

```diff
@@ -5,5 +5,5 @@ test('Sync Response fails due to invalid response type', async () => {
   const exampleSyncRes = require('./example.sync.response.json');
   const syncErrors = testlib.validate(exampleSyncRes, 'fail');
   console.log(syncErrors);
-  expect(syncErrors && syncErrors.join('\n')).toBe(undefined);
-});
\ No newline at end of file
+  expect(syncErrors && syncErrors.join('\n')).toThrow(TypeError);
```

Comment:
Does this allow the test to pass? It seems to me that the `validate` method is where the error is thrown and execution halts on line 6, so you'd never get that point.

Reply:
The test shouldn't pass. Because it specifies the argument type in the validation function, I think it will always hit that error every time a parameter string is not a sync, query, etc. Would it be more beneficial to not specify the response type in the validation function, so it can throw the error at the 'expect()...' part instead?

Reply:
All of the tests need to pass. The contents of a test may include throwing errors or checking false behavior. But the purpose of a unit test is to define and assert consistent behavior from a codebase. You can nest statements like `expect(testlib.validate...).toThrow()`.

Reply:
Got it! This makes sense now, pushed changes for the tests here

Reply:
Do we still need this file if the same test is run in `example.test.ts`?